src/components/Header/SearchHeader.tsx

```diff
@@ -62,10 +44,14 @@ const SearchHeader = ({
               }}
               isNarrowMobile={isNarrowMobile}
             >
-              <GlobalSelector
-                extendRight={true}
-                handleChange={handleSelectChange}
-              />
+              <ParentSize>
+                {() => (
+                  <SearchAutocomplete
+                    locations={getSearchAutocompleteLocations(region)}
+                    region={region}
+                  />
+                )}
```

Comment:
I think that the props `locations` and `region` are a bit redundant here, since the `locations` can be determined from the `region`. Consider either moving the logic to get the locations inside the component or not passing the `region` (I think that this might be better)